src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -218,4 +252,38 @@ class CallActor(key: CallKey, locallyQualifiedInputs: CallInputs, backend: Backe
   private def registerAbortFunction(abortFunction: AbortFunction): Unit = {
     self ! CallActor.RegisterCallAbortFunction(abortFunction)
   }
+
+  private def subscribe(subscriberActorRef: ActorRef, backendActor: ActorRef) = {
+    val subscription = Subscription[ActorRef](new ExecutionEvent, subscriberActorRef)
+    backendActor ! BackendActor.SubscribeToEvent[ActorRef](subscription)
+  }
+
+  // Currently this method will create a backend
+  private def createBackendActor(): ActorRef = {
+    val backendName: String = call.task.runtimeAttributes.attrs.get("backendName").headOption.map(_.head).getOrElse("")
+    val backendConfig = BackendConfiguration.apply()
+    val defaultBackendCfgEntry = backendConfig.getDefaultBackend()
+    val backendCfgEntry = backendConfig.getAllBackendConfigurations().filter(_.name.equals(backendName)).headOption.getOrElse(defaultBackendCfgEntry)
+    val task = buildTaskDescriptor()
+    backendCfgEntry.`type` match {
+      //TODO: Should be config based
+      case "ActorBased" => DefaultBackendFactory.getBackend(backendCfgEntry.initClass, actorSystem, task)
+      case oops => throw new IllegalStateException(s"Unknown type of backend defined: $oops! Only ActorBased allowed!")
+    }
+  }
+
+  /**
+    * Returns a TaskDescriptor, given the context of a task
+    *
+    * @return
+    */
+  private def buildTaskDescriptor(): TaskDescriptor = {
+    val name = call.fullyQualifiedName
+    val user = System.getProperty("user.name")
+    // Need Declarations, CallInputs, command template sequence
+    val cmdTemplateSeq = call.task.commandTemplate
+    val declarations = call.task.declarations
+    val runtimeAttributes = call.task.runtimeAttributes.attrs.map{case (k,v) => (k,v.head)}
```

Comment:
This is combining the HOF style with a dot prior to map. Also spaces on the insides of the braces. And what happens if `v.head` blows up?